aws_jobs/cyhy-data-extract.py

```diff
@@ -111,18 +113,23 @@ def cleanup_old_files(output_dir, file_retention_num_days):
 
 
 # TODO Finish function to delete files until there is only X in the bucket
-def cleanup_bucket_files(aws_access_key_id, aws_secret_access_key):
+def cleanup_bucket_files():
     """Delete oldest file if there are more than ten files in bucket_name."""
-    s3 = boto3.client(
-        "s3",
-        aws_access_key_id=aws_access_key_id,
-        aws_secret_access_key=aws_secret_access_key,
-    )
+    s3 = boto3.client("s3")
+
+    while True:
+        # Retrieve a list of applicable files.
+        ret = s3.list_objects_v2(Bucket=BUCKET_NAME, Prefix=SAVEFILE_PREFIX)
+        obj_list = ret["Contents"]
 
-    if len(s3.list_objects(Bucket=BUCKET_NAME)["Contents"]) > MAX_ENTRIES:
-        for key in s3.list_objects(Bucket=BUCKET_NAME)["Contents"]:
-            print(key)
-            print(key["LastModified"])
+        obj_list.sort(key=lambda x: x["Key"], reverse=True)
+        del_list = obj_list[MAX_ENTRIES:]
+
+        for obj in del_list:
```

Comment:
Can you use [`delete_objects`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.delete_objects) to make this delete more efficient?

Reply:
Yes, I can use `delete_objects` instead, however I would need to add manual error handling because it will fail silently otherwise. Using `delete_object` a ClientError exception is raised if there is an issue. When using `delete_objects` we instead get an array of errors and no exception is thrown.

Reply:
If the error handling isn't too burdensome it's probably worth doing.  If you're deleting a hundred files it will be much more efficient to make a single HTTPS API call instead of a hundred.